### ISSUE TITLE
StatePollingWait not correctly set when using Slurm

### ISSUE THREAD

**Issue by gmloose** [DataBiosphere/toil#3770]
This issue is related to #3768. There [it was mentioned ](https://github.com/DataBiosphere/toil/issues/3768#issuecomment-913632636) that if you do not explicitly set a state polling wait interval, using the command-line option `--statePollingWait`, that `getWaitDuration()` will be called. This, however, is not the case, as I mentioned in [my reaction](https://github.com/DataBiosphere/toil/issues/3768#issuecomment-913857405).

I debugged the issue a bit more, and I've come to the conclusion that the reason that `getWaitDuration()` is not called is that `config.statePollingWait` is set to `1` in two places in `config.py`:
https://github.com/DataBiosphere/toil/blob/1393068bd77f99aeae9faa9a10e20b54396b13e0/src/toil/common.py#L96 and https://github.com/DataBiosphere/toil/blob/1393068bd77f99aeae9faa9a10e20b54396b13e0/src/toil/common.py#L409
If I remove the `default=1` in line 409, and if I initialize `self.statePollingWait` to `0` in line 96, then `self.boss.config.statePollingWait` in https://github.com/DataBiosphere/toil/blob/1393068bd77f99aeae9faa9a10e20b54396b13e0/src/toil/batchSystems/abstractGridEngineBatchSystem.py#L54 evaluates to `False`, and `getWaitDuration()` will be called.

I don't know if these changes have impact on other configurations, though. So this may not be the correct fix.

**Comment by mr-c** [comment]
@gmloose Thanks for tracking this down! Can you send us a pull request with theses changes?

**Comment by adamnovak** [comment]
It makes sense to remove the defaults from the argument parsing and let the batch system provide the default, the way you suggest.

We might want to say `self.statePollingWait: Optional[Union[float, int]] = None` for the config.

**Comment by gmloose** [comment]
One question. With my fix, including @adamnovak's suggestion, I noticed that the `config.statePollingWait` is properly set when I'm in `_runMainLoop` in `toil/common.py:1149`, but it is reset to `None` when I'm in `workerScript` in `toil/worker.py:299`. I'm not sure if that's a problem. Maybe not, because it looks like Toil is now querying the Slurm accounting service at a much lower rate.

**Comment by gmloose** [comment]
Also, I see that a separate `sacct` command is issued for each submitted job. Even with an increased polling interval, this still leads to "hammering" the Slurm accounting service. Would it be possible to rewrite the logic so that the list of submitted jobs is passed in a single call of `sacct`, e.g. `sacct -n -j 711192,711193,711194 --format JobID,State,ExitCode -P -S 1970-01-01`. This of course requires some changes in how the output is parsed.

**Comment by mr-c** [comment]
> Would it be possible to rewrite the logic so that the list of submitted jobs is passed in a single call of `sacct`

Yes, this was recently done for the LSF backend https://github.com/DataBiosphere/toil/pull/3680 ; a pull request for the Slurm equivalent of that would be very welcome! You should be able to re-use the `--coalesceStatusCalls` command line option and implement `coalesce_job_exit_codes` in https://github.com/DataBiosphere/toil/blob/master/src/toil/batchSystems/slurm.py
